rclpy/src/rclpy/_rclpy.c

```diff
@@ -319,6 +384,15 @@ rclpy_get_ready_subscriptions(PyObject * Py_UNUSED(self), PyObject * args)
   return sub_ready_list;
 }
 
+/// Wait until timeout is reached or event happened
+/*
+ * \param[in] pywait_set Capsule pointing to the waitset structure
+ * \param[in] timeout optional, time to wait in nanoseconds
+ *                negative value means wait forever
+ *                zero means no wait
+ *                positive value: time to wait in nanoseconds
```

Comment:
This looks pretty awkward. I'd suggest explaining the different values in the body of the doc string and just leave this to something like "optional time to wait before waking up".